klpbuild/klplib/file2config.py

```diff
@@ -97,6 +97,17 @@ def find_configs_for_files(cs, file_paths: list):
         # Do not check headers
         if path.endswith('h'):
             continue
+
+        # Detect code that is only enabled on a specific architecture
+        if path.startswith("arch"):
+            if "s390" in path:
+                configs[path] = {'conf': "CONFIG_S390", 'module': 'vmlinux'}
+            elif "x86" in path:
+                configs[path] = {'conf': "CONFIG_X86_64", 'module': 'vmlinux'}
+            elif "powerpc" in path:
+                configs[path] = {'conf': "CONFIG_PPC64", 'module': 'vmlinux'}
+            continue
```

Comment:
I think we can make it a bit more robust by doing it just after the `_find_config()` and:
```suggestion
        if path.startswith("arch") and len(cs.get_all_configs(conf)) > 1:
            if "s390" in path:
                configs[path] = {'conf': "CONFIG_S390", 'module': 'vmlinux'}
            elif "x86" in path:
                configs[path] = {'conf': "CONFIG_X86_64", 'module': 'vmlinux'}
            elif "powerpc" in path:
                configs[path] = {'conf': "CONFIG_PPC64", 'module': 'vmlinux'}
            continue
```
So if the found `conf` is too generic, fallback to the arch specific one. What do you think? :)

Reply:
done